src/function/scalar/list/list_resize.cpp

```diff
@@ -0,0 +1,211 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/execution/expression_executor_state.hpp"
+
+#include <iostream>
+
+namespace duckdb {
+
+struct StandardCopyValue {
+	template <class T>
+	static void Operation(T *result_child_data, T *child_entries, optional_ptr<Vector>, Vector &, idx_t i,
+	                      idx_t result_child_offset) {
+		result_child_data[result_child_offset] = child_entries[i];
+	}
+};
+
+struct StringCopyValue {
+	template <class T>
+	static void Operation(T *result_child_data, T *child_entries, optional_ptr<Vector>, Vector &result_child, idx_t i,
+	                      idx_t result_child_offset) {
+		result_child_data[result_child_offset] = StringVector::AddString(result_child, child_entries[i]);
+	}
+};
+
+struct NestedCopyValue {
+	template <class T>
+	static void Operation(T *, T *, optional_ptr<Vector> child, Vector &result_child, idx_t i,
+	                      idx_t result_child_offset) {
+		result_child.SetValue(result_child_offset, child->GetValue(i));
+	}
+};
+
+template <class T, class COPY_FUNCTION>
+static void TemplatedListResizeFunction(DataChunk &args, Vector &result) {
+	D_ASSERT(result.GetType().id() == LogicalTypeId::LIST);
+	auto count = args.size();
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+
+	auto &list = args.data[0];
+	optional_ptr<Vector> child = &ListVector::GetEntry(args.data[0]);
+	auto &new_size = args.data[1];
+
+	UnifiedVectorFormat list_data;
+	UnifiedVectorFormat new_size_data;
+	UnifiedVectorFormat child_data;
+	list.ToUnifiedFormat(count, list_data);
+	new_size.ToUnifiedFormat(count, new_size_data);
+	child->ToUnifiedFormat(count, child_data);
+	auto list_entries = (list_entry_t *)list_data.data;
+	auto new_size_entries = (int64_t *)new_size_data.data;
+	auto child_entries = (T *)child_data.data;
+
+	// Find the new size of the result child vector
+	auto new_child_size = 0;
```

Comment:
We should explicitly set the type here to `idx_t` and ensure we correctly copy the new list sizes into it (see binding also).